src/test/java/seedu/address/model/appointment/AppointmentContainsDatePredicateTest.java

```diff
@@ -0,0 +1,123 @@
+                .build();
+        assertTrue(predicate.test(p));
+
+        // only start occurs in the given date
+        predicate = new AppointmentContainsDatePredicate(LocalDate.parse("23-10-2024", DATE_FORMATTER));
+        p = new PersonBuilder()
+                .withAppointment("Surgery",
+                        LocalDateTime.parse("23-10-2024-12-00", DATE_TIME_FORMATTER),
+                        LocalDateTime.parse("24-10-2024-12-00", DATE_TIME_FORMATTER))
+                .build();
+        assertTrue(predicate.test(p));
+
+        // only end date occurs in the given date
+        predicate = new AppointmentContainsDatePredicate(LocalDate.parse("24-10-2024", DATE_FORMATTER));
+        p = new PersonBuilder()
+                .withAppointment("Surgery",
+                        LocalDateTime.parse("23-10-2024-12-00", DATE_TIME_FORMATTER),
+                        LocalDateTime.parse("24-10-2024-12-00", DATE_TIME_FORMATTER))
+                .build();
+        assertTrue(predicate.test(p));
+
+        // date is within the start and end date
+        predicate = new AppointmentContainsDatePredicate(LocalDate.parse("24-10-2024", DATE_FORMATTER));
+        p = new PersonBuilder()
+                .withAppointment("Hospitalization",
+                        LocalDateTime.parse("23-10-2024-12-00", DATE_TIME_FORMATTER),
+                        LocalDateTime.parse("27-10-2024-12-00", DATE_TIME_FORMATTER))
+                .build();
+        assertTrue(predicate.test(p));
+
+        // null is given as the date
+        predicate = new AppointmentContainsDatePredicate(null);
+        p = new PersonBuilder()
+                .withAppointment("Surgery",
+                        LocalDateTime.parse("23-10-2024-12-00", DATE_TIME_FORMATTER),
+                        LocalDateTime.parse("24-10-2024-12-00", DATE_TIME_FORMATTER))
+                .build();
+        assertTrue(predicate.test(p));
+    }
+
+    @Test
+    public void test_appointmentContainsDate_returnsFalse() {
+        // Appointment does not exist with valid date input
+        AppointmentContainsDatePredicate predicate =
+                new AppointmentContainsDatePredicate(LocalDate.parse("23-10-2024", DATE_FORMATTER));
+        assertFalse(predicate.test(new PersonBuilder().build()));
+
+        // Appointment does not exist with null date input
+        predicate = new AppointmentContainsDatePredicate(null);
+        assertFalse(predicate.test(new PersonBuilder().build()));
+
+        // Non-matching dates
+        predicate = new AppointmentContainsDatePredicate(LocalDate.parse("01-01-2023", DATE_FORMATTER));
+        Person p = new PersonBuilder()
+                .withAppointment("Surgery",
+                        LocalDateTime.parse("23-10-2024-12-00", DATE_TIME_FORMATTER),
+                        LocalDateTime.parse("23-10-2024-14-00", DATE_TIME_FORMATTER))
+                .build();
+        assertFalse(predicate.test(p));
+    }
```

Comment:
Can perhaps include another test case with the Predicate date being AFTER the appointment, to cover all partitions.